### PR TITLE
OGM-775 Create an adaptor from JBoss Logging to Neo4J StringLogger

### PR DESCRIPTION
**Pull request by emmanuelbernard**
Used for query execution today

### CODE REVIEW

**Comment by Sanne** [comment]
I was curious to understand more about this, but I see no tests. Not worth it or are they coming next?

**Comment by emmanuelbernard** [comment]
No I wasn't planning on doing any test log.
To be fair, I've never seen the logs that @DavideD mentioned in my consoles.

**Comment by gunnarmorling** [comment]
Instead of feeding the execution engine's log statements through our logger, shouldn't we rather make sure that the execution engine uses the same configuration as the `GraphDatabaseService`?

The latter uses a default configuration which causes messages to go to _<DATABASE_DIR>/messages.log. Having parts of Neo4j messages not there but in "our" log seems confusing to me.

Unfortunately it seems not possible for the execution engine to "inherit" the logging config from the database service. So we may

* resemble that default config by doing the same what `InternalAbstractGraphDatabase#createLogging()` does (requires some tracing where that config originally is created during the bootstrap)
* OR expose a setting for the `Logging` implementation (and `isDebugEnabled()`) 

pass in a user-customizable EmbeddedGraphDatabaseFactory.create()

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/impl/StringLoggerToJBossLoggingAdaptor.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+/**
+ * Implements Neo4J's StringLogger and write the content to JBoss Logging.
+ * Regular messages are sent as debug, user messages as info, exception messages as error
+ * and long messages as separated calls to logMessage.
+ * This follows Neo4J's {@code LogbackService} implementation.
+ *
+ * @author Emmanuel Bernard &lt;emmanuel@hibernate.org&gt;
+ */
+public class StringLoggerToJBossLoggingAdaptor extends StringLogger {
```
Comment (gunnarmorling) [inline neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/impl/StringLoggerToJBossLoggingAdaptor.java]: If you want to go down that path, also `isDebugEnabled()` should be overridden (delegating to `log`) so one can actually make the messages visible.

**Comment by gunnarmorling** [comment]
@emmanuelbernard Any thoughts on my comment above?

**Comment by emmanuelbernard** [comment]
@gunnarmorling yes, I am only coming back to it now. I have other things to do today due to the trip but I'll try and address it.

**Comment by emmanuelbernard** [comment]
@gunnarmorling I think I found an acceptable solution. See second commit.
I also added the isDebugEnabled overriding you mentioned.

**Comment by gunnarmorling** [comment]
Merged, thanks!

**Pull request merged**